### ISSUE TITLE
Continous learning example fails with higher learning rate

### ISSUE THREAD

**Issue by JakeTurnick** [asieradzk/RL_Matrix#28]
![Image](https://github.com/user-attachments/assets/f58fce70-502b-471d-9f0f-b8bc9ffce0f9)

PPO agent with LR: 1E-02f fails partway through training with the error mentioned above, higher learning rate causes the crash sooner (1e-01f was much quicker) - slower learning rates have no issue

### Reproduce by - Take the Continuous learning example and increase the learning rate

Error details:
```
System.Runtime.InteropServices.ExternalException
  HResult=0x80004005
  Message=probability tensor contains either `inf`, `nan` or element < 0
Exception raised from multinomial_out at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\native\Distributions.cpp:608 (most recent call first):
00007FF9E47238F200007FF9E4723890 c10.dll!c10::Error::Error [<unknown file> @ <unknown line number>]
00007FF9E4722C8A00007FF9E4722C30 c10.dll!c10::detail::torchCheckFail [<unknown file> @ <unknown line number>]
00007FF9B757135900007FF9B7570E70 torch_cpu.DLL!at::native::multinomial_out [<unknown file> @ <unknown line number>]
00007FF9B7570DE000007FF9B7570CA0 torch_cpu.DLL!at::native::multinomial [<unknown file> @ <unknown line number>]
00007FF9390DFD4E00007FF9390B9130 torch_cuda.DLL!at::cuda::where_outf [<unknown file> @ <unknown line number>]
00007FF939002F1A00007FF938F96A80 torch_cuda.DLL!at::cuda::bucketize_outf [<unknown file> @ <unknown line number>]
00007FF9B7A8F50F00007FF9B7A71E90 torch_cpu.DLL!at::TensorMaker::make_tensor [<unknown file> @ <unknown line number>]
00007FF9B7B16CC400007FF9B7A71E90 torch_cpu.DLL!at::TensorMaker::make_tensor [<unknown file> @ <unknown line number>]
00007FF9B7BFA8DD00007FF9B7BFA7F0 torch_cpu.DLL!at::_ops::multinomial::redispatch [<unknown file> @ <unknown line number>]
00007FF9B92143B000007FF9B916A7D0 torch_cpu.DLL!torch::autograd::CopySlices::~CopySlices [<unknown file> @ <unknown line number>]
00007FF9B91E0F5900007FF9B916A7D0 torch_cpu.DLL!torch::autograd::CopySlices::~CopySlices [<unknown file> @ <unknown line number>]
00007FF9B7A8F50F00007FF9B7A71E90 torch_cpu.DLL!at::TensorMaker::make_tensor [<unknown file> @ <unknown line number>]
00007FF9B7A8F3E200007FF9B7A71E90 torch_cpu.DLL!at::TensorMaker::make_tensor [<unknown file> @ <unknown line number>]
00007FF9B7B843D500007FF9B7B842F0 torch_cpu.DLL!at::_ops::multinomial::call [<unknown file> @ <unknown line number>]
00007FF9B720C9B900007FF9B720C960 torch_cpu.DLL!at::Tensor::multinomial [<unknown file> @ <unknown line number>]
00007FF9E51AAFEB00007FF9E51AAF50 LibTorchSharp.DLL!THSTensor_multinomial [<unknown file> @ <unknown line number>]
00007FF985109EEB <unknown symbol address> !<unknown symbol> [<unknown file> @ <unknown line number>]

  Source=TorchSharp
  StackTrace:
   at TorchSharp.torch.CheckForErrors()
   at TorchSharp.torch.Tensor.multinomial(Int64 num_samples, Boolean replacement, Generator generator)
   at TorchSharp.torch.multinomial(Tensor input, Int64 num_samples, Boolean replacement, Generator generator)
   at RLMatrix.Agents.PPO.Implementations.PPOActionSelection`1.SelectDiscreteActionsFromProbs(Tensor result, Int32[] actionSize)
   at RLMatrix.Agents.PPO.Implementations.ContinuousPPOAgent`1.SelectActions(T[] states, Boolean isTraining)
   at RLMatrix.Agents.PPO.Implementations.LocalContinuousPPOAgent`1.SelectActionsBatchAsync(List`1 stateInfos, Boolean isTraining)
   at RLMatrix.Agents.Common.LocalContinuousRolloutAgent`1.<>c__DisplayClass4_0.<<Step>b__0>d.MoveNext()
   at RLMatrix.Agents.Common.LocalContinuousRolloutAgent`1.<Step>d__4.MoveNext()
   at Program.<<Main>$>d__0.MoveNext() in E:\Programming\ML\RLMatrix\RLMatrix Tutorial\RLMatrixTutorial\Program.cs:line 27

```

**Comment by asieradzk** [comment]
Hi. This is not a bug and something that can happen in deep reinforcement learning. If you happen to update gradients too drastically the will start to produce outputs beyond float32 range. 

Best way to avoid this is to give penalties for outputs that are too large or too small and of course use more gentle learning rate, smaller num of epochs or larger batch size, or gentler reward signals.

**Comment by JakeTurnick** [comment]
Ok - thanks for the info, I'll keep that in mind as I play around with this
